src/capture.rs

```diff
@@ -27,8 +27,17 @@ use {trace, resolve, SymbolName};
 #[derive(Clone)]
 //~ HACK1 #[derive(RustcDecodable, RustcEncodable)]
 //~ HACK2 #[derive(Deserialize, Serialize)]
-pub struct Backtrace {
-    frames: Vec<BacktraceFrame>,
+pub enum Backtrace {
```

Comment:
Unfortuntely the way visibility works means that the internal details of this are exposed, but perhaps a representation like:

```rust
pub struct Backtracke {
    inner: Inner,
}

enum Inner {
    Resolved { ... }
    Unresolved { ... }
}
```